gpflow/mean_functions.py

```diff
@@ -72,6 +68,24 @@ def __call__(self, X):
         return tf.matmul(X, self.A) + self.b
 
 
+class Identity(Linear):
+    """
+    y_i = x_i
+    """
+    def __init__(self, input_dim=None):
+        Linear.__init__(self)
+        del self.A
+        del self.b
+        self.A = Parameter(np.eye(input_dim), trainable=False)
+        self.b = Parameter(np.zeros(input_dim), trainable=False)
+        if input_dim is None:
```

Comment:
This will never be true, as `np.eye()` is going to throw an exception when `input_dim` is None. You can either default to 1, as the `Linear` class, or simply make the input_dim argument non-optional, then you don't need the check either.